### PR TITLE
Add filters

### PR DESCRIPTION
**Pull request by obourdon**
Add filter option

Running hssh with wildcards allows to select proper host depending
on filter criteria. For instance:

`HSSH_FILTER=instance:_1 HSSH_INTERACTIVE=0 AWS_PROFILE=XXX hssh core@coreos_test_env-dev_* hostname`

will select host whose fields with name `'.*instance.*'` match `'.*_1.*'`

If no matching host is found, hssh exits with error.
If multiple matching hosts are found, hssh exits with error.
If no filter is used, 1st host in list is used.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -58,16 +62,76 @@ func handleJump(args []string, provider string) []string {
 	return ssh_args_str
 }
 
-func selectHost(hosts []cr.Host) cr.Host {
+func hostMatch(f *cr.Host, filterstr []string) bool {
```
Comment (jeromer) [inline main.go:69]: This function is too complex

File: main.go [hunk 2]
```diff
@@ -58,16 +62,76 @@ func handleJump(args []string, provider string) []string {
...
+				log.Tracef("Field name: %s not matching %s", typeField.Name, filterstr[0])
+			}
+		}
+	}
+	return false
+}
+
+func filterHosts(hosts []cr.Host, filter string) cr.Host {
+	host := hosts[0]
+	if filter != "" {
```
Comment (jeromer) [inline main.go]: How about doing `if filter == "" { return host}` and keep to rest of the code not a big branch ?

File: main.go (hunk 1 above)
Comment (obourdon) [reply main.go:69]: Added more comments but not sure can be made clearer any other way ;-)

File: main.go (hunk 2 above)
Comment (obourdon) [reply main.go]: Done

**Review by jeromer — APPROVED** [review]

**Pull request merged**